net/authz/validators.go

```diff
@@ -0,0 +1,47 @@
+package authz
+
+import "context"
+
+type key int
+
+const (
+	tokenKey key = iota
+	localhostKey
+)
+
+func authzToken(ctx context.Context) bool {
+	// TODO(tessr): compare against Policies
+	_, ok := TokenFromContext(ctx)
+	return ok
+}
+
+func authzLocalhost(ctx context.Context) bool {
+	// TODO(tessr): compare against Policies
+	return LocalhostFromContext(ctx)
+}
+
+// NewContextWithToken sets the token in a new context and returns the context.
+func NewContextWithToken(ctx context.Context, token string) context.Context {
+	return context.WithValue(ctx, tokenKey, token)
+}
+
+// TokenFromContext returns the token stored in the context, if there is one.
+func TokenFromContext(ctx context.Context) (string, bool) {
```

Comment:
I'd say these accessors and the `key` type belong in package authn, and authz should depend on authn rather than the other way around. Your user name (your identity) is part of and defined by authentication, it is used by authorization.

I think this would also mean the NewContextFoo functions can be un-exported inside of authn.

As a guess, I bet that having the dependency go from authn->authz is to avoid a dependency from something outside of http on something inside of that directory? I'd say maybe let's not worry about it for now or just move this package in there as well. I can think of a bunch of tricks we might do to pull more code out of that subdirectory but not sure if it's worth it rn.